document/js-api/index.bs

```diff
@@ -1329,6 +1383,165 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
+        1. Let |args| be the result of [=coerce JavaScript arguments|coercing arguments=] (|functype|,|arguments|)
+        1. Let |map| be the [=surrounding agent=]'s associated [=Execution Context Status map=].
+        1. Let |ec| be the currently executing [=execution context=], i.e., the [=execution context=] that is at the top of the [=surrounding agent=]'s current [=execution context stack=].
+        1. Assert: |map| does not contain any entry for |ec|.
+        1. Add an entry mapping |ec| to [=active=] in |map|.
+        1. Let (|store|, |result|) be the result of [=func_invoke=](|store|, |funcaddr|, |args|).
+        1. Assert: if control reaches here, we have done waiting for suspended imports
+        1. If the entry for |ec| in |map| is not [=active=] then trap. Otherwise, remove the entry for |ec| from [=map=].
+        1. Set the [=surrounding agent=]'s [=associated store=] to |store|.
+        1. If |result| is [=error=], throw an exception. This exception should be a WebAssembly {{RuntimeError}} exception, unless otherwise indicated by <a href="#errors">the WebAssembly error mapping</a>.
+        1. Otherwise, if |result| is of the form [=throw=] exnaddr,
+            1. Perform [=EvaluateCall=] (|promise|.\[[Reject]],|result|.,false).
+        1. Otherwise, assert |result| is a [=list=] of WebAssembly values
+            1. Let |outArity| be the [=list/size=] of |result|.
+            1. If |outArity| is 0, return undefined.
+            1. Otherwise, if |outArity| is 1, let |jsReturnValue| be [=ToJSValue=](|result|[0]).
+            1. Otherwise,
+               1. Let |values| be « ».
+               1. [=list/iterate|For each=] |r| of |result|,
+               1. [=list/Append=] [=ToJSValue=](|r|) to |values|.
+               1. let |jsReturnValue| be [$CreateArrayFromList$](|values|).
+            1. Perform [=EvaluateCall=] (|promise|.\[[Resolve]],|jsReturnValue|,false)
+    1. Let |con| be [=CreateBuiltinFunction=](|runner|,0,"",<<>>)
+    1. Perform [$Call$](|con|, undefined, <<>>)
+    1. Returns |promise|
+</div>
+
+Note: The extra |$Call$| in the above algorithm ensures that the creation of the Promise is separated from the fullfilling of that Promise. In effect, this allows suspension of the fullfillment to occur whilst allowing the creation of the Promise itself to continue.
+
+<div algorithm>
+    The <dfn constructor for="Suspending">Suspending(|jsFun|)</dfn> constructor, when invoked, performs the following steps:
+    1. If [$IsCallable$](|jsFun|) is false, throw a {{TypeError}}.
+    1. Let |suspendingProto| be \[[%WebAssembly.Suspending.prototype%]]
+    1. Let |susp| be the result of [$OrdinaryObjectCreate$](|suspendingProto|)
+    1. Assign the \[[wrappedFunction]] internal slot of |susp| to |jsFun|
+    1. Return |susp|
+</div>
+
+<div algorithm>
+To <dfn>create a suspending function</dfn> from a JavaScript function |func|, with type |functype| perform the following steps:
```

Comment:
You referred to these above as "Suspending objects", not functions.

Reply:
The Suspending object refers to the result of using new WebAssembly.Suspending on a JavaScript function.
A suspending function is the function that is created during the module instantiation process -- using the Suspending object as the trigger.